tools/update-copyright-notices.sh

```diff
@@ -6,8 +6,7 @@ echo "Updating global copyright strings..."
 sed -re 's/(Copyright.*2[0-9]{3}-)[0-9]{4}/\1'"${YEAR}"'/g' -i src/res/translation/*.ts src/util.cpp src/aboutdlgbase.ui
 
 echo "Updating copyright comment headers..."
-for file in $(find android ios linux mac src windows -regex '.*\.\(cpp\|h\|mm\)' -not -regex '\./\(\.git\|libs/\|moc_\|ui_\).*'); do
-		sed -re 's/(\*.*Copyright.*[^-][0-9]{4})(\s*-\s*\b[0-9]{4})?\s*$/\1-'"${YEAR}"'/' -i "${file}"
-done
+find android ios linux mac src windows '(' -regex '.*\.\(cpp\|h\|mm\)' -not -regex '\./\(\.git\|libs/\|moc_\|ui_\).*' ')' -exec \
+    sed -re 's/(\*.*Copyright.*[^-][0-9]{4})(\s*-\s*\b[0-9]{4})?\s*$/\1-'"${YEAR}"'/' -i {} +
```

Comment:
This looks really unreadable in my opinion...

Reply:
I agree. Any suggestions? We could move the sed pattern to a variable, but I'm not sure if it gains us much?

Reply:
Why is the for loop not ok?

Reply:
```
In tools/update-copyright-notices.sh line 9:
for file in $(find android ios linux mac src windows -regex '.*\.\(cpp\|h\|mm\)' -not -regex '\./\(\.git\|libs/\|moc_\|ui_\).*'); do
            ^-- SC2044 (warning): For loops over find output are fragile. Use find -exec or a while read loop.
```

We could add the `while` loop instead. I think I'd rather go with the single-command solution (`find ... -exec`).

```bash
find android ios linux mac src windows -regex '.*\.\(cpp\|h\|mm\)' -not -regex '\./\(\.git\|libs/\|moc_\|ui_\).*') | while read -r file; do
    sed -re 's/(\*.*Copyright.*[^-][0-9]{4})(\s*-\s*\b[0-9]{4})?\s*$/\1-'"${YEAR}"'/' -i "${file}"
done

Reply:
It's looks still a bit messy, but maybe a bit better?

Reply:
Ok, changed.